metasync.js

```diff
@@ -72,3 +72,28 @@ metasync.sequential = function(funcs, done) {
   if (len > 0) next();
   else done();
 };
+
+
+// Asynchrous filter
+// filter :: (a -> Boolean) -> [a] -> ([a] -> Void)
+metasync.filter = function(predicate, array, done) {
```

Comment:
According to [Filter article on Wikipedia](https://en.wikipedia.org/wiki/Filter_(higher-order_function)) most of languages use `filter(predicate, array)` order of arguments rather than `filter(array, predicate)`